### DISCUSSION TITLE
GOlang PDK: Response() phase is not being executed during HTTP/2  request

### DISCUSSION THREAD

**Discussion by upvest-mike** [Kong/kong#8492] [Help] [answered]
Hello!
I have a custom plugin written on Go.  This plugin uses Access(...) and Response(...) phases to access  requests and responses data. While the plugin is getting  requests by HTTP/1.1  protocol - everything  works fine. Access(...) and Response(...) phases are executed, but when plugin gets requests by HTTP/2 protocol, Response(...) phase stop executing.
I investigates a little and found this code 
https://github.com/Kong/kong/blob/c324715838e0eeaeed513361d54677d8ec26e220/kong/init.lua#L869
Do I understand correctly that for the HTTP/2 protocol Kong skips Response(...) phase? And if I wrong, then how to setup plugin and get the Response(...) phase executed with HTTP/2?

Kong version: 2.4.0

Thanks.

**Comment by StarlightIbuki** [comment, marked as answer]
It seems that the response phase is not supported for HTTP/2. You can use `header_filter` and `body_filter` instead to implement the same logic.
